forte/helpers/ndarray/ndarray.hpp

```diff
@@ -0,0 +1,399 @@
+#pragma once
+
+#include <complex>
+#include <iostream>
+#include <memory>
+#include <span>
+#include <vector>
+
+#include <pybind11/numpy.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include "datatype.hpp"
+
+namespace py = pybind11;
+using namespace pybind11::literals;
+
+constexpr bool ndarray_debug = true;
+
+namespace forte {
+
+/// @brief Class to represent a tensor
+template <typename T> class ndarray {
+    /// @brief Shape of tensor
+    using shape_t = std::vector<size_t>;
```

Comment:
Mirroring `np.empty` or `np.zeros`, should we change `shape_t` to `std::tuple`?

Reply:
That's a good suggestion, but unfortunately we don't template over the number of dimensions (should we?), so this has to be a dynamically allocated quantity (std::tuple has compile-time dimension fixed).

Reply:
Good point! I still think front a user's perspective `ndarray([3,4])` looks like it should return a 1-D array of length 2 set to 3 and 4, instead of a 2-D empty array with 12 elements. So it might be worth just doing a pybind type conversion from tuple to list before passing to C++?

Reply:
I see what you mean. I now have enabled passing the shape via a tuple. The list is disabled. To make the class accept a list with data, I would need to write code to handle arbitrary cases, which is not simple. Since the point of this object is to facilitate Py<->C++ interoperability, it won't reproduce the full extent of capabilities of ndarrays in np. If the user wants to create an array from a list or list of lists of numbers, they can do so in numpy and then create a C++ ndarray from the python array.